### ISSUE TITLE
IDE slowly gets less and less responsive and eventually crashes

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#20093]
I am editing a big C# file. IDE gets slower and slower to respond and eventually crashes. I sent several Watson dumps. 
I also have a dump taken when VS is using close to 2GB of memory, it looks like there might be a memory "leak" somewhere. I captured a dump, it is available upon request.

CC @Pilchie

**Comment by CyrusNajmabadi** [comment]
Tagging @sharwell .  Can you make your dump available Aleksey?

**Comment by CyrusNajmabadi** [comment]
Can you also provide information about teh version of VS you're using, and what version of the Roslyn insiders vsix you're using if you've updated your vsix.  Thanks!

**Comment by AlekseyTs** [comment]
```
Microsoft Visual Studio Enterprise 2017 Int Preview
Version 15.3.0 Preview 1.0 [26526.1.d15rel]
VisualStudio.15.IntPreview/15.3.0-pre.1.0+26526.1.d15rel
Microsoft .NET Framework
Version 4.6.01055

Installed Version: Enterprise

Architecture Diagrams and Analysis Tools   00369-60000-00001-AA300
Microsoft Architecture Diagrams and Analysis Tools

Visual Basic 2017   00369-60000-00001-AA300
Microsoft Visual Basic 2017

Visual C# 2017   00369-60000-00001-AA300
Microsoft Visual C# 2017

Visual C++ 2017   00369-60000-00001-AA300
Microsoft Visual C++ 2017

Visual F# 4.1   00369-60000-00001-AA300
Microsoft Visual F# 4.1

Application Insights Tools for Visual Studio Package   8.8.00518.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2017   15.0.30523.0
ASP.NET and Web Tools 2017

ASP.NET Core Razor Language Services   1.0
Provides languages services for ASP.NET Core Razor.

ASP.NET Template Engine 2017   15.0.30523.0
ASP.NET Template Engine 2017

ASP.NET Web Frameworks and Tools 2012   4.0.20506.0
For additional information, visit https://www.asp.net/

ASP.NET Web Frameworks and Tools 2017   5.2.50511.0
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   15.0.30522.0
Azure App Service Tools v3.0.0

Azure Data Lake Node   1.0
This package contains the Data Lake integration nodes for Server Explorer.

Azure Data Lake Tools for Visual Studio   2.2.7000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Data Lake Tools for Visual Studio   2.2.7000.0
Microsoft Azure Data Lake Tools for Visual Studio

Common Azure Tools   1.9
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Dotfuscator Community Edition   5.30.0.5167-c98e2f8
PreEmptive Protection - Dotfuscator CE

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

JavaScript UWP Project System   2.0
JavaScript UWP Project System

MenuCommands Extension   1.0
MenuCommands Visual Studio Extension Detailed Info

Microsoft Azure HDInsight Azure Node   2.2.7000.0
HDInsight Node under Azure Node

Microsoft Azure Hive Query Language Service   2.2.7000.0
Language service for Hive query

Microsoft Azure Stream Analytics Language Service   2.2.7000.0
Language service for Azure Stream Analytics

Microsoft Azure Stream Analytics Node   1.0
Azure Stream Analytics Node under Azure Node

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2017 - v2.9.50519.1

Microsoft Continuous Delivery Tools for Visual Studio   0.3
Simplifying the configuration of continuous build integration and continuous build delivery from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NuGet Package Manager   4.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

SQL Server Data Tools   15.1.61702.140
Microsoft SQL Server Data Tools

TestDriven.Net   4.0-beta3
Zero friction testing for Visual Studio

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript   2.2.0.0
TypeScript tools for Visual Studio

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio tools for CMake   1.0
Visual Studio tools for CMake

Visual Studio Tools for Universal Windows Apps   15.0.26526.01
The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.
```

Nothing extra on top.

**Comment by sharwell** [comment]
This appears to be a duplicate of #20065, at least in the sense that #20096 fixes it. I see 696 instances of `WorkspaceAnalyzerOptions` holding on to 452 `Solution` instances.

**Comment by mavasani** [comment]
This should be fixed with https://github.com/dotnet/roslyn/pull/20096, which has been merged now.

**Comment by sharwell** [comment]
@AlekseyTs I have now triaged what I believe to be the major issues revealed by the process dump provided for this issue, with the dominating problem now resolved. If #20094 was a crash dump *from the same process*, we can go ahead and close it as a duplicate (the hang would have been caused by extreme GC overhead as we approached the OOM condition). If it is not from the same process, I will investigate that issue separately.
